lib/torrent.js

```diff
@@ -281,83 +278,78 @@ class Torrent extends EventEmitter {
   }
 
   _onListening () {
-    const self = this
-    if (self.discovery || self.destroyed) return
+    if (this.discovery || this.destroyed) return
 
-    let trackerOpts = self.client.tracker
+    let trackerOpts = this.client.tracker
     if (trackerOpts) {
-      trackerOpts = extend(self.client.tracker, {
-        getAnnounceOpts () {
+      trackerOpts = extend(this.client.tracker, {
+        getAnnounceOpts: () => {
           const opts = {
-            uploaded: self.uploaded,
-            downloaded: self.downloaded,
-            left: Math.max(self.length - self.downloaded, 0)
+            uploaded: this.uploaded,
+            downloaded: this.downloaded,
+            left: Math.max(this.length - this.downloaded, 0)
           }
-          if (self.client.tracker.getAnnounceOpts) {
-            extendMutable(opts, self.client.tracker.getAnnounceOpts())
+          if (this.client.tracker.getAnnounceOpts) {
+            extendMutable(opts, this.client.tracker.getAnnounceOpts())
           }
-          if (self._getAnnounceOpts) {
+          if (this._getAnnounceOpts) {
             // TODO: consider deprecating this, as it's redundant with the former case
-            extendMutable(opts, self._getAnnounceOpts())
+            extendMutable(opts, this._getAnnounceOpts())
           }
           return opts
         }
       })
     }
 
     // begin discovering peers via DHT and trackers
-    self.discovery = new Discovery({
-      infoHash: self.infoHash,
-      announce: self.announce,
-      peerId: self.client.peerId,
-      dht: !self.private && self.client.dht,
+    this.discovery = new Discovery({
+      infoHash: this.infoHash,
+      announce: this.announce,
+      peerId: this.client.peerId,
+      dht: !this.private && this.client.dht,
       tracker: trackerOpts,
-      port: self.client.torrentPort,
+      port: this.client.torrentPort,
       userAgent: USER_AGENT
     })
 
-    self.discovery.on('error', onError)
-    self.discovery.on('peer', onPeer)
-    self.discovery.on('trackerAnnounce', onTrackerAnnounce)
-    self.discovery.on('dhtAnnounce', onDHTAnnounce)
-    self.discovery.on('warning', onWarning)
```

Comment:
Changed these to in-line arrow functions instead of being hoisted. IMO it's actually more readable that way, but it's a change in style.